### ISSUE TITLE
how to test on my custom dataset?

### ISSUE THREAD

**Issue by sepid91** [HamadYA/GhostFaceNets#30]
Hi, 
I want to test GhostFace net on my own dataset, But unfortunately i cannot find the way.
Do i need to train this network on my dataset? 
what file i need for evaluate this code?

**Comment by HamadYA** [comment]
> Hi, I want to test GhostFace net on my own dataset, But unfortunately i cannot find the way. Do i need to train this network on my dataset? what file i need for evaluate this code?

Hi,

You do not need to train the network on your dataset if the task is face verification.

You can use:
eval_folder.py
to run model evaluation on any custom dataset folder, which is in the same format with Training dataset.

.               # dataset folder
├── 0           # person folder
│   ├── 100.jpg # face image
│   ├── 101.jpg # face image
│   └── 102.jpg # face image
├── 1           # person folder
│   ├── 111.jpg
│   ├── 112.jpg
│   └── 113.jpg
├── 10
│   ├── 707.jpg
│   ├── 708.jpg
│   └── 709.jpg

**Comment by sepid91** [comment]
thanks for your response, 
I made this dataset and named each folder by the name of the person but i didn't get how to make this file "Evaluating bin files"?
and for testing i run the below line.
python ./eval_folder.py -d {DATA_PATH} -m {BASIC_MODEL.h5}

Is that the true way for testing? how can i give my testing image?

**Comment by HamadYA** [comment]
> I made this dataset and named each folder by the name of the person but i didn't get how to make this file "Evaluating bin files"?

Sorry, I did not fully understand the question, do you want to convert it to .bin file?

For testing on a directory, you can use:
python ./eval_folder.py -d {DATA_PATH} -m {BASIC_MODEL.h5}

And for:
how can i give my testing image?

I also did not understand it.
Sorry

Can you please elaborate more.

**Comment by sfxeazilove** [comment]
for my own case i tried to modify the eva folder to some face evaluation on just two images, basically. like the code below

import os
import numpy as np
import tensorflow as tf
from skimage.io import imread
from tensorflow.keras.preprocessing.image import img_to_array, load_img
from sklearn.preprocessing import normalize

# Set up GPU configuration
# gpus = tf.config.experimental.list_physical_devices("GPU")
# for gpu in gpus:
#     tf.config.experimental.set_memory_growth(gpu, True)

def load_and_preprocess_image(img_path, target_size=(112, 112)):
    """Load and preprocess an image."""
    img = load_img(img_path, target_size=target_size)
    img = img_to_array(img)
    img = (img - 127.5) * 0.0078125
    return img

def compute_similarity(model, img1_path, img2_path):
    """Compute similarity between two images using the model."""
    
    # Load and preprocess images
    img1 = load_and_preprocess_image(img1_path)
    img2 = load_and_preprocess_image(img2_path)
    
    # Predict embeddings
    emb1 = model.predict(np.expand_dims(img1, axis=0))
    emb2 = model.predict(np.expand_dims(img2, axis=0))
    
    # Normalize embeddings
    emb1 = normalize(emb1)
    emb2 = normalize(emb2)
    
    # Compute similarity
    similarity = np.dot(emb1, emb2.T)
    print(similarity)
    
    return similarity[0][0]

if __name__ == "__main__":
    folder_path = "image_test"
    model_path = "checkpoints/GhostFaceNet_W1.3_S1_ArcFace.h5"
    
    # Check there are exactly 2 images in the folder
    images = [f for f in os.listdir(folder_path) if f.lower().endswith(('.png', '.jpg', '.jpeg'))]
    if len(images) != 2:
        print("The folder should contain exactly 2 images.")
        exit(1)
    
    # Load the model
    model = tf.keras.models.load_model(model_path)
    
    # Compute the similarity
    img1_path = os.path.join(folder_path, images[0])
    img2_path = os.path.join(folder_path, images[1])
    similarity = compute_similarity(model, img1_path, img2_path)
    
    print(f"Similarity between {images[0]} and {images[1]}: {similarity:.4f}")

is there something i am not doing right? Because the similarity score when i used similar images in the image_test path are not quite different for non similar images, (Recall, i am just passing two images at the same time in the path, either similar or unsimilar) this is quite hard to get a threshold.
i understood yolov5 was used for face detection, does that mean i have to download it from its source repo and call the yolov5 class. that would mean, i have two model files, the yolov5 and the Ghostfacenets.

1. How can you help to modify this evaluation script to fit GhostfaceNets intended purpose
2. is there a particular threshold for GhostFaceNets?

**Comment by HamadYA** [comment]
Closed for now, if you still have question re-open it please.
